packages/amplify-provider-awscloudformation/lib/configuration-manager.js

```diff
@@ -357,44 +357,39 @@ function removeProjectConfig(context) {
 }
 
 async function loadConfiguration(context, awsClient, attatchRegion) {
-  process.env.AWS_SDK_LOAD_CONFIG = true;
   const projectConfigInfo = getCurrentConfig(context);
-  if (projectConfigInfo.configLevel !== 'general') {
+  if (projectConfigInfo.configLevel === 'project') {
     const { config } = projectConfigInfo;
+    let awsconfig;
     if (config.useProfile) {
-      process.env.AWS_PROFILE = config.profileName;
-      const credentials = new awsClient.SharedIniFileCredentials({
-        profile: config.profileName,
-      });
-      awsClient.config.credentials = credentials;
+      awsconfig = await systemConfigManager.getProfiledAwsConfig(config.profileName);
     } else {
-      awsClient.config.loadFromPath(config.awsConfigFilePath);
+      awsconfig = JSON.parse(fs.readFileSync(config.awsConfigFilePath, 'utf8'));
     }
+    awsClient.config.update(awsconfig);
+  } else if (attatchRegion) {
+    awsClient.config.update({ region: getRegion() });
   }
-  if (attatchRegion) {
-    awsClient.region = getRegion(projectConfigInfo);
-  }
+
   return awsClient;
 }
 
-function getRegion(projectConfigInfo) {
+function getRegion() {
   // For details of how aws region is set, check the following link
   // https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/setting-region.html
-  if (projectConfigInfo.configLevel === 'general') {
-    if (process.env.AWS_REGION) {
-      return process.env.AWS_REGION;
-    }
-    if (process.env.AMAZON_REGION) {
-      return process.env.AMAZON_REGION;
-    }
-    const profileName = process.env.AWS_PROFILE || 'default';
-    return systemConfigManager.getProfile(profileName).region;
+
+  if (process.env.AWS_REGION) {
+    return process.env.AWS_REGION;
+  }
+  if (process.env.AMAZON_REGION) {
+    return process.env.AMAZON_REGION;
   }
-  const { config } = projectConfigInfo;
-  if (config.useProfile) {
-    return systemConfigManager.getProfile(config.profileName).region;
+  const profileName = process.env.AWS_PROFILE || 'default';
+  const profileRegion = systemConfigManager.getProfileRegion(profileName);
+  if (profileRegion) {
+    return profileRegion;
   }
-  return config.region;
+  return 'us-east-1';
```

Comment:
Why should it always return us-east-1?

Reply:
If it's 'general', shouldn't the region be present in either the env variables or the default?

Reply:
removed default and throw error if region can not be resolved.

Reply:
I refactored this part. 
I removed attach region flag for the method, and let the caller (aws-pinpoint.js) handle the specifics, this is now much cleaner.